Use appendAction for translated HeaderValueOptions (#8678)
solo-io/gloo: docs/content/guides/traffic_management/request_processing/append_remove_headers/_index.md

4. Now, requests that match the route `/a/1` get the following headers:
   * The `x-gateway-start-time` request header is inherited from the parent VirtualHost option.
   * The `x-route-table` request header is set in the child Route option.

>    * The `x-route-table` response header in the parent overwrites the child object's value of `a` instead to `alphabet`, because child objects take precedence in case of conflict.

Reviewer: > because child objects take precedence in case of conflict.

Should this now say the parent object takes precedence?

Reviewer: or maybe we just remove that last part and let the description you added after explain it?
```suggestion
   * The `x-route-table` response header in the parent overwrites the child object's value of `a` instead to `alphabet`.
```

Due to how header manipulations are processed, less specific headers overwrite more specific headers.

From the [Envoy docs](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/headers#custom-request-response-headers):
> Headers are appended to requests/responses in the following order: weighted cluster level headers, route level headers, virtual host level headers and finally global level headers.

> In our above example with `x-route-table`, since the virtual host level header is evaluated after the route level header, the virtual host level header overwrites the route level header.

Reviewer: ```suggestion
In the previous example of the `x-route-table` response header, the virtual host level header overwrites the route level header because the virtual host level header is evaluated after the route level header.
```

> If we had set `append: true` on the virtual host, or removed it, then the route level header would have been appended to the virtual host level header.

Reviewer: By "removed it" do you mean remove the `append` field altogether? like not have it at all as in the original example?

Reviewer: Could we also provide an example of what appending would look like?
```suggestion
If you set `append: true` or omit this field on the virtual host, then the route level response header (`a`) would get appended to the virtual host level header (`alphabet`), such as `x-route-table: alphabeta`.
```

Author: Ah yeah good callout, good to say `omit`!

As far as what the expected response, I'm not quite sure. I know using `curl`, appending leads to two headers being returned, each with the individual value.

_But_ I also saw [this issue](https://github.com/solo-io/gloo/issues/8634) that is pointing to the likelyhood that the above is broken. and instead of multiple of the same header, it should be `x-route-table: alphabet a` (for our example).

I left it a bit vague just referencing an append occuring, since I'm not sure if we have a bug causing this to append wonkily.

Reviewer: ok, yeah we can leave out the example if we have bugs, and just add the expected outcome after that is fixed.

> If we had set `append: false` on the route, since the route is evaluated before the virtual host, it would not affect the virtual host.

Reviewer: similar comment
```suggestion
If you set `append: false` on the route, the route is evaluated before the virtual host and does not affect the virtual host. In the example, the response header would stay `x-route-table: alphabet`.
```

Author: Do you think we could we keep the `since` or add a `because`? 

The reason this behaves the way it does is _because_ of the route is evaluated before virtual host. Without the "since", at least to me, it sounds like this evaluation is due to `append: false`, which would be wrong.

Reviewer: Yeah, that makes sense! I would just say to use because instead of since (since is usually for time, like since five minutes ago).

Author: Ah makes sense, thank you!
